avm/res/db-for-my-sql/flexible-server/main.bicep

```diff
@@ -68,8 +68,8 @@ param geoRedundantBackup string = 'Enabled'
 @description('Optional. The mode to create a new MySQL server.')
 param createMode string = 'Default'
 
-@description('Conditional. The managed identity definition for this resource. Required if \'customerManagedKey\' is not empty.')
-param managedIdentities managedIdentitiesType
+@description('Optional. The managed identity definition for this resource.')
+param userAssignedIdentities object = {}
```

Comment:
Hey @FallenHoot,
be aware this is breaking with several AVM specs (most notably [this one](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#managed-identities)). The interface was defined in the way it is to allow or a consistent user experience accross all AVM modules.

I appreciate your comment that for this module it may be an overkill, but in that case I'd rather strongly advice to define a new spec, than to diverge completely.

On that note also one question:
- The original implementation allowed the user to define an array of user-assiged identities via their resource ID
- The new implementation only allows for exact one user assigned identity
Does the resource only allow for one single identity?

Again, I'd rather like to keep the consistency high and if only one single identity is allowed, stick to the same user-defined type be it only with the change to allow for a single resource ID, and only adjust all the ther pieces.

Before I add even more to this comment, let me wait for your reply and then let's go from there 😄 

cc: @Azure/avm-module-reviewers-bicep

Reply:
It works as multiple objects as well. I never changed how the test worked, just the naming, but if it is breaking the AVM specs, I will try to revert what it is now.

Reply:
Hey @FallenHoot,
please excuse the late reply. Let me look into this in the next couple days - I just returned from some time off.

Reply:
Hey @FallenHoot,
I can't find the reason for the `Expected $null or empty, because required parameters in the template file should have a description that starts with "Required.". Found incorrect items: [managedIdentities.userAssignedResourceIds]., but got 'managedIdentities.userAssignedResourceIds'.` error as the `userAssignedResourceIds` property of the corresponding type looks absolutely fine to me. 
Anyways - considering the latest state of the module in `main` this should not be an issue anymore anyways as the local UDT was replaced with a reference from the `avm-common-types` utility module:
https://github.com/Azure/bicep-registry-modules/blob/701814f4db319874a3912f0cfd4bc9f3b0160c25/avm/res/db-for-my-sql/flexible-server/main.bicep#L72-L74

I'd recommend pulling in the latest changes from main as at least the reason for this comment should be gone now :)